components/public-api/gitpod/v1/workspace.proto

```diff
@@ -65,6 +72,51 @@ message ListWorkspacesResponse {
   PaginationResponse pagination = 2;
 }
 
+message CreateAndStartWorkspaceRequest {
+  message Source {
+
+    message Git {
+      string clone_url = 1;
+
+      // ref is an alternatively symbolic. e.g. refs/tags/v1.0,
+      // 9585191f37f7b0fb9444f35a9bf50de191beadc2
+      string ref = 2;
+
+      // create_local_branch is the branch you want to create based on provided
+      // clone_url and ref when workspace started
+      string create_local_branch = 3;
+    }
+
+    Git git = 1;
+  }
+
+  string organisation_id = 1;
+
+  string project_id = 2;
+
+  Source source = 3;
+
+  repeated WorkspaceEnvironmentVariable additional_env_variables = 4;
```

Comment:
worth adding docs about that they are overriding any env vars contributed through user and configuration env vars